pkgs/development/compilers/rust/rustc.nix

```diff
@@ -1,25 +1,26 @@
 { stdenv, removeReferencesTo, pkgsBuildBuild, pkgsBuildHost, pkgsBuildTarget
 , fetchurl, file, python3
-, llvm_9, darwin, cmake, rust, rustPlatform
+, llvm, darwin, cmake, rust, rustPlatform
 , pkgconfig, openssl
 , which, libffi
 , withBundledLLVM ? false
 , enableRustcDev ? true
 , version
 , sha256
 , patches ? []
+, preConfigure ? null
 }:
 
 let
   inherit (stdenv.lib) optionals optional optionalString;
   inherit (darwin.apple_sdk.frameworks) Security;
 
-  llvmSharedForBuild = pkgsBuildBuild.llvm_9.override { enableSharedLibraries = true; };
```

Comment:
I do not think this change here is wise. This likely breaks cross-compilation.

Reply:
hmm indeed, I didn't notice this. Should I replace this with a "selector" function, like `pkgs: pkgs.llvm_7`?

Reply:
I am not entirely sure I can see what you are saying. Can you draft an example here?

Reply:
https://gist.github.com/ProgVal/da945eab4ec04f55123411b4063de001

It's a pattern I noticed these files already use (`selectRustPackage`)